Library/Homebrew/rubocops/cask/stanza_order.rb

```diff
@@ -17,34 +17,38 @@ class StanzaOrder < Base
 
         def on_cask(cask_block)
```

Comment:
@issyl0, what I meant was that this cop shouldn't be run `on_cask`, but rather there should be something like `on_cask_stanza_block` instead, which matches `cask` and `on_*` blocks equally and this cop is then implemented in terms of `on_cask_stanza_block` rather than `on_cask`. So the cop itself doesn't have to loop through nested blocks.

No idea though if that makes a difference in how autocorrect behaves though.

Reply:
Unfortunately I still don't understand, sorry. How would it "treat cask and on blocks equally" when one is within the other? How to get to that state?

Reply:
I added a commit that adds `on_cask_stanza_block`, maybe that makes it clearer.

Reply:
I thought it did help, but then I tried to use it and was confused again. I guess I don't understand how to make use of it and which bits it skips out having to do in the cop itself?

Reply:
> which bits it skips out having to do in the cop itself

Looping through nested blocks.

Reply:
> how to make use of it

I assume most of the needed methods are defined in `CaskBlock` which isn't used in this case.